pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -62,18 +62,16 @@ var (
 	accountsLock = &sync.Mutex{}
 )
 
-func newBlobDiskController(common *controllerCommon) (*BlobDiskController, error) {
-	c := BlobDiskController{common: common}
-
-	// get accounts
-	accounts, err := c.getAllStorageAccounts()
-	if err != nil {
-		klog.Errorf("azureDisk - getAllStorageAccounts error: %v", err)
-		c.accounts = make(map[string]*storageAccountState)
-		return &c, nil
+func (c *BlobDiskController) initStorageAccounts() {
+	if c.accounts == nil {
```

Comment:
Are there any race conditions for it? maybe hold the operations within a lock?

Reply:
@feiskyer good catch, while I think it does not matter since it's only get accounts(read), no write or update.

Reply:
Yep, but I think `c.accounts` may be called in different goroutines, so it's better to be hold by a lock.

Reply:
ok, let me put a lock anyway for the sake of safety, thanks.

Reply:
@feiskyer it's been fixed in the new commit